packages/expo-modules-core/ios/JSI/EXJavaScriptWeakObject.mm

```diff
@@ -13,9 +13,14 @@ @implementation EXJavaScriptWeakObject {
   __weak EXJavaScriptRuntime *_runtime;
 
   /**
-   Shared pointer to the `WeakRef` JS object.
+   Shared pointer to the `WeakRef` JS object. Available only on JSC engine.
    */
-  std::shared_ptr<jsi::Object> _jsObject;
+  std::shared_ptr<jsi::Object> _weakRef;
+
+  /**
+   A weak reference to a JS object. Available only on Hermes engine.
+   */
+  std::shared_ptr<jsi::WeakObject> _weakObject;
```

Comment:
do you want to have `#if __has_include(<reacthermes/HermesExecutorFactory.h>)` also to these member variables? and you can also use the same `_weakRef` name

e.g.

```c++
#if __has_include(<reacthermes/HermesExecutorFactory.h>)
  /**
   A weak reference to a JS object. Available only on Hermes engine.
   */
  std::shared_ptr<jsi::WeakObject> _weakRef;
#else
  /**
   Shared pointer to the `WeakRef` JS object. Available only on JSC engine.
   */
  std::shared_ptr<jsi::Object> _weakRef;
#endif
```

Reply:
Good idea! I applied your suggestion, but used `_weakObject` instead of `_weakRef` because the latter has more sense when it refers to `WeakRef` instance.